core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostEstimator.scala

```diff
@@ -73,14 +85,21 @@ class DefaultCostEstimator(
     case Some(_) =>
   }
 
-  override def estimate(region: Region, resourceUnits: Int): Double = {
-    this.operatorEstimatedTimeOption match {
+  override def allocateResourcesAndEstimateCost(
+      region: Region,
+      resourceUnits: Int
+  ): (Region, Double) = {
+    // Currently the dummy cost from resourceAllocator is discarded.
+    val (newRegion, _) = resourceAllocator.allocate(region)
```

Comment:
we might want to change the return of `resourceAllocator.allocate` to just resourceConfigs, instead of an updated region.